src/cmd-release-ocp

```diff
@@ -0,0 +1,167 @@
+#!/usr/bin/env python3
+# NOTE: PYTHONUNBUFFERED is set in the entrypoint for unbuffered output
+#
+# Publish an oscontainer as the machine-os-content in an OpenShift release
+# series.
+
+import atexit
+import argparse
+import json
+import logging as log
+import os
+import shutil
+import sys
+import tarfile
+import tempfile
+import urllib.request
+
+
+COSA_PATH = os.path.dirname(os.path.abspath(__file__))
+sys.path.insert(0, COSA_PATH)
+from cosalib.meta import GenericBuildMeta as Meta
+from cosalib.cmdlib import (
+    get_basearch,
+    run_verbose
+)
+
+
+OCP_SERVER = "https://api.ci.openshift.org"
+OCP_RELEASE_STREAM = "quay.io/openshift-release-dev/ocp-release"
+OCP_TOOL_MIRROR = "https://mirror.openshift.com/pub/openshift-v4/clients/oc/{ocp_ver}/linux/oc.tar.gz"
+
+log.basicConfig(
+    format='[%(levelname)s]: %(message)s',
+    level=log.INFO)
+
+
+def ocp_versions(meta):
+    """
+    Return the corresponding OCP version.
+    RHCOS is versioned <OCP MAJ><OCP MIN>.<RHEL MAJ><RHEL MIN>.
+    For example OCP version 4.6 with RHEL 8.1 would be 46.81
+    """
+
+    if "OpenShift" not in meta.get("summary", None):
+        return None, None
+    ostree_v = meta.get("ostree-version").split('.')
+    ocp_major, ocp_minor = list(ostree_v[0])
+    os_major, os_minor = list(ostree_v[1])
+    return (f"{ocp_major}.{ocp_minor}", f"{os_major}.{os_minor}")
+
+
+def fetch_ocp_bin(ocp_ver):
+    """
+    Download the specific release oc binary
+    """
+    url = OCP_TOOL_MIRROR.format(ocp_ver=ocp_ver)
+    log.info(f"Downloading oc tool from {url}")
+
+    oc_gz = tempfile.NamedTemporaryFile()
+    oc_bin = tempfile.NamedTemporaryFile()
+
+    with urllib.request.urlopen(url) as data:
+        oc_gz.write(data.read())
```

Comment:
This buffers the whole binary to memory, you want `shutil.copyfile()` as you also use below.